File: problem/Keep_Multiplying/findFinalValue.py
class Solution:
    def findFinalValue(self, nums, original):
        bits = 0
        for n in nums:
            if n % original != 0:
                continue
            n //= original
            if (n & (n - 1)) == 0:
                bits |= n
        
        bits += 1
        return original * (bits & -bits)

File: problem/Keep_Multiplying/findFinalValue.c
int findFinalValue(int *nums, int numsSize, int original)
{
    int bits = 0;
    for (int i = 0; i < numsSize; i++)
    {
        int n = nums[i];
        if (n % original != 0)
            continue;
        n /= original;
        if ((n & (n - 1)) == 0)
            bits |= n;
    }

    bits++;
    return original * (bits & -bits);
}